cmd/postcli/main.go

```diff
@@ -74,6 +77,9 @@ func parseFlags() {
 	flag.StringVar(&commitmentAtxIdHex, "commitmentAtxId", "", "commitment atx id, in hex (required)")
 	numUnits := flag.Uint64("numUnits", uint64(opts.NumUnits), "number of units")
 
+	flag.UintVar(&nonces, "nonces", 16, "number of nonces for -genproof")
```

Comment:
The number of nonces must be a multiple of 16 (a nonce group). How about changing this argument into "number of nonce groups" to avoid setting an invalid value? Also, 16 is unrealistic, perhaps a more typical value like `288` would be better?
```suggestion
	flag.UintVar(&noncesGroups, "nonceGroups", 18, "number of nonce groups for -genproof. Each group consists of 16 nonces.")
```
And multiply it by 16 later in the code.

OR at least inform the user that it must be a multiple of 16:
```suggestion
	flag.UintVar(&nonces, "nonces", 16, "number of nonces for -genproof. Must be a multiple of 16.")
```